test.js

```diff
@@ -0,0 +1,252 @@
+import test from 'node:test'
+import http from 'node:http'
+import { createHandler } from './index.js'
+import Redis from 'ioredis'
+import { once } from 'node:events'
+import assert from 'node:assert/strict'
+import * as ethers from 'ethers'
+
+let signer
+let server
+let redis
+let api
+
+test.before(async () => {
+  signer = ethers.Wallet.createRandom()
+  const logger = {
+    error: console.error,
+    info: console.info,
+    request: () => {}
+  }
+
+  redis = new Redis()
+  await redis.flushall()
+  const handler = await createHandler({
+    logger,
+    redis,
+    signerAddresses: [await signer.getAddress()]
+  })
+  server = http.createServer(handler)
+  server.listen()
+  await once(server, 'listening')
+  api = `http://127.0.0.1:${server.address().port}`
+})
+
+test.after(() => {
+  server.close()
+  redis.disconnect()
+})
+
+const sign = async participants => {
+  const digest = ethers.solidityPackedKeccak256(
+    ['address[]', 'uint256[]'],
+    [Object.keys(participants), Object.values(participants)]
+  )
+  const signed = await signer.signMessage(digest)
+  const { v, r, s } = ethers.Signature.from(signed)
+  return { v, r, s }
+}
+
+test('scheduled rewards', async t => {
+  // Tests rely on the state created by each other. This is a shortcut and
+  // should eventually be improved.
+
+  await t.test('empty scheduled rewards', async t => {
+    {
+      const res = await fetch(`${api}/scheduled-rewards`)
+      assert.deepEqual(await res.json(), {})
+    }
+    {
+      const res = await fetch(`${api}/log`)
+      assert.deepEqual(await res.json(), [])
+    }
+  })
+  await t.test('set scores', async t => {
+    {
+      const scores = {
+        '0x000000000000000000000000000000000000dEaD': '10',
+        '0x000000000000000000000000000000000000dEa2': '100'
+      }
+      const res = await fetch(`${api}/scores`, {
+        method: 'POST',
+        body: JSON.stringify({
+          scores,
+          signature: await sign(scores)
+        })
+      })
+      assert.strictEqual(res.status, 200)
+      assert.deepStrictEqual(await res.json(), {
+        '0x000000000000000000000000000000000000dEaD': '4566',
+        '0x000000000000000000000000000000000000dEa2': '45662'
+
+      })
+    }
+    {
+      const res = await fetch(`${api}/scheduled-rewards`)
+      assert.deepEqual(await res.json(), {
+        '0x000000000000000000000000000000000000dEaD': '4566',
```

Comment:
Smart contract's [setScore()](https://github.com/Meridian-IE/impact-evaluator/blob/14f630fb871f9b2e90bbcc8ad8b3e34c98d6ab2f/src/ImpactEvaluator.sol#L140-L143) method discards scores for the burner address. Should we do the same in spark-rewards?

Reply:
great catch! done